static-assets/components/cstudio-forms/forms-engine.js

```diff
@@ -2394,6 +2395,12 @@ const initializeCStudioForms = () => {
 								tinymce.get(rteId).remove();
 							});
 						}
+						// When rendering the items of a repeat group, if there are RTE fields we need to clear beforeSaveCallbacks
+						// to avoid having multiple callbacks for the same RTE field (since callbacks are going to be added on each
+						// RTE rendering)
+						controlEl.form.beforeSaveCallbacks = controlEl.form.beforeSaveCallbacks.filter(
```

Comment:
The other issue was that when re-rendering the items of a repeat group (for example when adding a new item), the RTE control sets a `beforeSaveCallback` to verify that the control is properly loaded before saving the form. What happened was that the callbacks we're not cleaned up on re-rendering, and in some cases that was causing the ready validation to fail. By cleaning it up before re-rendering the items (I added the repeat group container Id info into the callback for a precise filtering), we make sure the RTE callbacks are up to date.